Heart/Provider/Application/NewAccountByProvider.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Heart\Provider\Application;
+
+use Heart\Provider\Domain\DTOs\NewProviderDTO;
+use Heart\Provider\Domain\Entities\ProviderEntity;
+use Heart\Provider\Domain\Enums\ProviderEnum;
+use Heart\Provider\Domain\Repositories\ProviderRepository;
+use Heart\User\Domain\Repositories\UserRepository;
+
+class NewAccountByProvider
+{
+    public function __construct(
+        private readonly UserRepository $userRepository,
+        private readonly ProviderRepository $providerRepository,
+    ) {
+    }
+
+    public function handle(ProviderEnum $providerEnum, string $providerId, string $username): ProviderEntity
```

Comment:
Would be good to rename the `handle` method name, since it doesn't describes what it does. Using something like `findOrCreate` would be better IMO.